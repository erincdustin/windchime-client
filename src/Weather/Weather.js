import React from 'react';

function Weather(props) {
  let weather = {
    Temperature: {
      Metric: {
        Value: null,
      },
    HasPrecipitation: null,
    },
    CloudCover: null,
    PrecipitationType: null,
  }

  if(props.weather !== null) {
    weather = props.weather;
  }
  return (
    <div>
      <h3>Find current weather conditions:</h3>
      <span className="weather-btn">
          <button 
          onClick={()=> props.searchWeather()}
            className="btn btn-default" type="button">Get Weather</button>
      </span>
      <br />
      <div>Temp: {((weather.Temperature.Metric.Value) * (9/5)) + 32} degrees, 
            Precipitation: {weather.HasPrecipitation}, 
            Cloud Cover: {weather.CloudCover}, 
            PrecipitationType: {weather.PrecipitationType}
      </div>
    </div>
  );
}

export default Weather;