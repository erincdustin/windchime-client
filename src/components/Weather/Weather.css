@import url('https://fonts.googleapis.com/css?family=Open+Sans&display=swap');

#weather-form {
  font-family: 'Helvetica', 'Arial', sans-serif;
  text-align: center;
  padding: 25px 15px;
  background-color: rgb(243, 243, 243);
  border-radius: 10px;
}

input{
  width: 50%;
  padding: 8px 20px;
  margin: 8px 0;
  border: 1px solid #ccc;
  border-radius: 10px;
  box-sizing: border-box;
}