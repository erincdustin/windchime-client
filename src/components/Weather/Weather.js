import React from 'react';
import { Link, Redirect } from 'react-router-dom'

class Weather extends React.Component {

  render() {

  return (
    <div>
      <h3>Find current weather conditions:</h3>
      <form id="weather-form" onSubmit={(e)=> {
            e.preventDefault();
            this.props.searchCity(e.target.weatherInput.value);   
            this.props.history.push('/genreOption');  
          }}>
        <input id="weatherInput" name="weatherInput" className="weatherInput" type="text" placeholder="Enter a zip code" required></input>
        <button className="btn btn-default" type="submit">Get Weather</button>
      </form>
    </div>
  );
}
}

export default Weather;