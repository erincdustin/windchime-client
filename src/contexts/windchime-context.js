import React, { Component } from 'react'

const WindChimeContext = React.createContext({
    error: '',
    songs: null,
    weather: null,
    genreChoice: null,
    id: null,
    artists: null,
    topArtists: null,
    playlistId: null,
    userPlaylists: null,
    snapshot: null,
    targetEnergy: null,
    targetValence: null,
    targetTempo: null,
    targetPopularity: null,
    setError: () => { },
    clearError: () => { },
    setSongs: () => { },
    setWeather: () => { },
    setGenreChoice: () => { },
    setId: () => { },
    setArtists: () => { },
    setTopArtists: () => { },
    setPlaylistId: () => { },
    setUserPlaylists: () => { },
    setSnapshot: () => { },
    setTargetEnergy: () => { },
    setTargetValence: () => { },
    setTargetTempo: () => { },
    setTargetPopularity: () => { },
})

export default WindChimeContext;

export class WindChimeProvider extends Component {
    constructor(props) {
        super(props)
        const state = {
            error: '',
            songs: null,
            weather: null,
            genreChoice: null,
            id: null,
            artists: null,
            topArtists: null,
            userPlaylists: null,
            playlistId: null,
            snapshot: null,
            targetEnergy: null,
            targetValence: null,
            targetTempo: null,
            targetPopularity: null,
            setError: this.setError,
            clearError: this.clearError,
            setSongs: this.setSongs,
            setWeather: this.setWeather,
            setGenreChoice: this.setGenreChoice,
            setId: this.setId,
            setArtists: this.setArtists,
            setTopArtists: this.setTopArtists,
            setPlaylistId: this.setPlaylistId,
            setUserPlaylists: this.setUserPlaylists,
            setSnapshot: this.setSnapshot,
            setTargetEnergy: this.setTargetEnergy,
            setTargetValence: this.setTargetValence,
            setTargetTempo: this.setTargetTempo,
            setTargetPopularity: this.setTargetPopularity
          }   
        this.state = state; 
    }

    setError = error => {
      console.error(error)
      this.setState({ error })
    }

    clearError = () => {
      this.setState({ error: '' })
    }

    setUserPlaylists = (userPlaylists) => {
      this.setState({ userPlaylists })
    }

    setSongs = (songs) => {
      this.setState({ songs })
    }

    setWeather = (weather) => {
      this.setState({ weather })
    }

    setGenreChoice = (genreChoice) => {
      this.setState({ genreChoice })
    }

    setId = (id) => {
      this.setState({ id })
    }

    setArtists = (artists) => {
      this.setState({ artists })
    }

    setTopArtists = (topArtists) => {
      this.setState({ topArtists })
    }

    setPlaylistId = (playlistId) => {
      this.setState({ playlistId })
    }

    setSnapshot = (snapshot) => {
      this.setState({ snapshot })
    }

    setTargetEnergy = (targetEnergy) => {
      this.setState({ targetEnergy })
    }

    setTargetValence = (targetValence) => {
      this.setState({ targetValence })
    }

    setTargetTempo = (targetTempo) => {
      this.setState({ targetTempo })
    }

    setTargetPopularity = (targetPopularity) => {
      this.setState({ targetPopularity })
    }

    render() {
        return (
            <WindChimeContext.Provider value={this.state}>
                {this.props.children}
            </WindChimeContext.Provider>
        )
    }
}
