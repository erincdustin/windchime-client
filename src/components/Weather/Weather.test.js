import React from 'react';
import renderer from 'react-test-renderer';
import { library } from '@fortawesome/fontawesome-svg-core'
import { faWind } from '@fortawesome/free-solid-svg-icons'
import { WindChimeProvider } from '../../contexts/windchime-context';
import Weather from './Weather';

library.add(faWind)

describe(`Weather`, () => {
  it('renders as expected', () => {
    const tree = renderer
      .create(<WindChimeProvider><Weather /></WindChimeProvider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})