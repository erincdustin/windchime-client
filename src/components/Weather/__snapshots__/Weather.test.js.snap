// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather renders as expected 1`] = `
<div>
  <div
    className="ribbon one"
  >
    <div
      className="ribbon-header"
    >
      Weather
    </div>
  </div>
  <form
    id="weather-form"
    onSubmit={[Function]}
  >
    
    <input
      aria-invalid={false}
      aria-label="Zip code entry"
      aria-required="true"
      className="input"
      id="weatherInput"
      name="weatherInput"
      placeholder="Enter zip code"
      required={true}
      type="text"
    />
    <button
      className="btn btn-default"
      type="submit"
    >
      Get Weather
    </button>
  </form>
</div>
`;
